Develop --> Master | frontend

Yaroslav7125 opened the pull request:
Баханов Ярослав
https://etc4b.sse.codesandbox.io/

Yaroslav7125 committed:
Init project

Yaroslav7125 committed:
Add main layout for pages.

Yaroslav7125 committed:
Add main layout for pages and add font, static files.

Yaroslav7125 committed:
Merge branch 'feature/Creating_main_layout' into develop

Yaroslav7125 committed:
Add part with content activity page.

Yaroslav7125 committed:
Add activity page

Yaroslav7125 committed:
Add map page

Yaroslav7125 committed:
Add normal map

Yaroslav7125 committed:
Add normal map and style.

Yaroslav7125 committed:
Merge branch 'feature/Creating_map_page' into develop

Yaroslav7125 committed:
Add loading map with marker

Yaroslav7125 committed:
Add timer page.

Yaroslav7125 committed:
Add timer

Yaroslav7125 committed:
Add vuex store for logic timer

Yaroslav7125 committed:
Merge branch 'feature/Creating_timer' into develop

Yaroslav7125 committed:
Add  todo page and add components for todo page(AddTodo.vue, Todo.vue, TodoList.vue)

Yaroslav7125 committed:
Merge branch 'feature/Creating_todo_page' into develop

Yaroslav7125 committed:
Set up eslint config and rewrote wrong code

Yaroslav7125 committed:
Rewrote code by eslint warninngs

Yaroslav7125 committed:
Add eslint with webee plagins

Yaroslav7125 committed:
Add connections count and change textarea

Yaroslav7125 committed:
Add using Eslint config webbe

Yaroslav7125 committed:
Rewrote code from errors eslint.

Yaroslav7125 committed:
Add :key="todo.id"

Yaroslav7125 committed:
rename ChangeTodoComlited to changeTodoComplited

crutch12 commented on .eslintrc.js:
```diff
@@ -0,0 +1,30 @@
+    'plugin:@web-bee-ru/vue', // vue
+
+    // npm i --save-dev eslint-plugin-vue eslint-import-resolver-nuxt
+    'plugin:@web-bee-ru/nuxt', // nuxt (vue)
```
Тебе тут только `plugin:@web-bee-ru/nuxt` достаточно, он уже наследуется от vue, а тот от base

crutch12 commented on store/index.js:
```diff
@@ -0,0 +1,20 @@
+import Vue from 'vue';// eslint-disable-line
+import Vuex from 'vuex';// eslint-disable-line
```
nuxt вроде сам подключает Vuex и это лишнее

crutch12 replied:
`eslint-disable-line` лишний

crutch12 commented on pages/map.vue:
```diff
@@ -0,0 +1,81 @@
+      apiKey: 'AIzaSyC0dy2XqP-GPUFSTeGTKpkmT55k9GdHzxU',
+      version: 'weekly',
+    });
+    loader.load().then(() => { // eslint-disable-next-line
```
```
  .then((google) => {
    new google.maps.Map(document.getElementById("map"), mapOptions);
  })
  ```
  
  Вот так правильно - принимать google в then
  Сейчас у тебя вообще чудом работает, поэтому eslint жалуется

crutch12 replied:
https://www.npmjs.com/package/@googlemaps/js-api-loader

Тут же написано

Yaroslav7125 committed:
Delete lines:
 'plugin:@web-bee-ru/base',
 'plugin:@web-bee-ru/vue',

Yaroslav7125 committed:
Fix map load - then part

Yaroslav7125 committed:
fix map and delete lines:
 import Vue from 'vue';// eslint-disable-line
 import Vuex from 'vuex';// eslint-disable-line

Yaroslav7125 committed:
fix case

crutch12 commented on pages/map.vue:
```diff
@@ -0,0 +1,77 @@
+
+</script>
+
+<style>
```
Почему не scoped?

Yaroslav7125 replied:
Исправил в коммите 104b7b7.

crutch12 commented on pages/map.vue:
```diff
@@ -0,0 +1,77 @@
+          <img src="@/static/img/UpStroke.svg" alt="">
+        </div>
+      </div>
+      <div id="map"></div>
```
id в данном случае норм, но если бы ты оформил map.vue в компонент, то кто-то легко мог бы инициализировать 2 таких компонента, тогда у тебя id бы повторялся - это очень не хорошо
Лучше класс использовать

Yaroslav7125 replied:
Исправил в коммите e6870e8, чёт я не продумал то что в проект можно запихать несколько компонентов-карт)

crutch12 commented on pages/map.vue:
```diff
@@ -0,0 +1,77 @@
+</script>
+
+<style>
+#map{
```
Лучше класс (сверху писал)

crutch12 commented on pages/time.vue:
```diff
@@ -0,0 +1,51 @@
+  },
+};
+</script>
+<style>
```
Почему не scoped?

Yaroslav7125 replied:
Исправил в коммите 104b7b7.

crutch12 commented on pages/index.vue:
```diff
@@ -0,0 +1,347 @@
+
+<script>
+</script>
+<style scoped>
```
А почему scss не используешь? (везде)
Можно же написать `lang="scss"` и он даст писать в scss

Yaroslav7125 replied:
Переправил в коммите e88cf95

crutch12 commented on pages/map.vue:
```diff
@@ -0,0 +1,77 @@
+          lat: crd.latitude,
+          lng: crd.longitude,
+        });
+        // eslint-disable-next-line
```
Лишний коммент

Yaroslav7125 replied:
Убрал в коммите d018d8c.

crutch12 commented on pages/todo.vue:
```diff
@@ -0,0 +1,94 @@
+
+</script>
+
+<style>
```
scoped

Yaroslav7125 replied:
Исправил в коммите 104b7b7.

crutch12 commented on static/css/font2.css:
```diff
@@ -0,0 +1,63 @@
+/* cyrillic-ext */
```
А зачем font2?

crutch12 replied:
Мне казалось, что ты смог подключить напрямую по ссылке гугла в итоге

crutch12 replied:
Поэтому непонятно, зачем эти css файлы вообще

Yaroslav7125 replied:
Исправил в коммите 2da067e.

crutch12 commented on static/css/main.css:
```diff
@@ -0,0 +1,5 @@
+body{
```
Лучше, конечно, делать main.scss
Т.к. любой css удобнее писать через scss :))

Yaroslav7125 replied:
Исправил в коммите 8760c4e. Он подписан как тсправление всего css, но там только один файл.

crutch12 commented on pages/index.vue:
```diff
@@ -0,0 +1,347 @@
+                <span>Himalayan sunset</span>
+              </div>
+              <div class="post-fp-right-part d-flex align-items-center">
+                <img src="@/static/img/DoneBadge.svg" alt="">
```
Если ты кладёшь картику в папку `static`, то она становится доступна напрямую по ссылке:
`sitename.ru/img/DoneBadge.svg`

Поэтому нету смысла дополнительно класть её в папку сгенерированных ассетов, добавляя `@/static`
(то есть сейчас правильно писать просто `/img/DoneBadge.svg`)

Вообще для ассетов правильнее делать так:
Класть их в папку assets (например)
И из неё уже забирать через `@/assets/...`

А вот если доступ к файлу нужен обязательно через `sitename.ru/...` (например robots.txt)
Тогда файл надо класть в папку static

Yaroslav7125 replied:
Исправил в коммите 1639c92

crutch12 commented on store/index.js:
```diff
@@ -0,0 +1,18 @@
+import Vuex from 'vuex';// eslint-disable-line
+
+const store = () => new Vuex.Store({// eslint-disable-line
```
коммент лишний

Yaroslav7125 replied:
Исправил в коммите d46bf34

crutch12 reviewed (commented):
Оставил комменты

Yaroslav7125 committed:
Add scoped to style.

Yaroslav7125 committed:
Исправил обращение обращение по id на обращение по классу.

Yaroslav7125 committed:
убрал лишний коммент на 35 строчке.

Yaroslav7125 committed:
Удалил неиспользуемые шрифты
 font1.css
 font2.css

Yaroslav7125 committed:
Переименовал пути к папке static при использовании сайтом картинок.

Yaroslav7125 committed:
Убрал лишнии коменты из файла vuex(index.js)

Yaroslav7125 committed:
Переписал всё на scss.

Yaroslav7125 committed:
Исправил стили в компонентах на scss.

Yaroslav7125 committed:
Удалил подключение 2gisApi

Yaroslav7125 committed:
Подправил css (отступ у таймера)

Yaroslav7125 committed:
Добавил запросы на бэк, и сортировку по id

Yaroslav7125 committed:
Добавил баг c изменением title и решил аналогичный c изменеием completed.
Решение - передача самого обьекта todo в событии

Yaroslav7125 committed:
Подправил код с именением completed todo

Yaroslav7125 committed:
Пофиксил баг с изменением title.

Yaroslav7125 committed:
Добавил BaseURL для зпросов на бэк,

Yaroslav7125 committed:
Исрпавил положенеи заголовка

Yaroslav7125 committed:
Добавил использование переменых среды
BASE_URL- baseUrд для запросов на backend

Yaroslav7125 committed:
Удалил неиспользумемые пакеты
